SecretSanta/src/SecretSanta.Data/Gift.cs

```diff
@@ -4,24 +4,15 @@
 
 namespace SecretSanta.Data
 {
-    public class Gift
+    public class Gift : FingerPrintEntityBase
     {
-        public int Id { get; }
         public string Title { get => _Title; set => _Title = value ?? throw new ArgumentNullException(nameof(Title)); }
         private string _Title = string.Empty;
         public string Description { get => _Description; set => _Description = value ?? throw new ArgumentNullException(nameof(Description)); }
         private string _Description = string.Empty;
         public string Url { get => _Url; set => _Url = value ?? throw new ArgumentNullException(nameof(Url)); }
         private string _Url = string.Empty;
-        public User User { get; set; }
-
-        public Gift(int id, string title, string description, string url, User user)
-        {
-            Id = id;
-            Title = title;
-            Description = description;
-            Url = url;
-            User = user;
-        }
+        public User User { get=>_User; set=>_User=value??throw new ArgumentNullException(nameof(User)); }
+        private User _User = null!;
```

Comment:
Consider just making the backing field a new user with a default constructor. It is generally not safe to initialize a backing field to null if you can't help it
```suggestion
        private User _User = new User();
```

Reply:
good point! I'll clean that up

Reply:
Actually I can't do that, it breaks tests on it because the created blank users are saved to the DB